diff --git a/provider/manganato/manganato.go b/provider/manganato/manganato.go
index e7d0c62..7af6542 100644
--- a/provider/manganato/manganato.go
+++ b/provider/manganato/manganato.go
@@ -20,7 +20,7 @@ var Config = &generic.Configuration{
 		query = strings.TrimSpace(query)
 		query = strings.ToLower(query)
 		query = url.QueryEscape(query)
-		template := "https://chapmanganato.com/https://manganato.com/search/story/%s"
+		template := "https://manganato.com/search/story/%s"
 		return fmt.Sprintf(template, query)
 	},
 	MangaExtractor: &generic.Extractor{
